src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -505,7 +505,6 @@ static void Main()
     {
         int x, y;
         var type = ((x, y) = new C()).GetType();
```

Comment:
I would expect to get a parsing error for this case. If I am not mistaken, we settled on the design where deconstruction is a statement, not an expression that can be used in other context.